velox/expression/ExprCompiler.cpp

```diff
@@ -92,12 +92,25 @@ struct Scope {
   }
 };
 
+// Utility method to check if inputs of an expression are of the same type which
+// allow it to be flattened.
+bool checkIfInputsAllowToFlatten(const TypedExprPtr& expr) {
```

Comment:
nit: perhaps, rename this function to allInputTypesEquivalent or similar to match the implementation. Then comment will become redundant or won't need to specify how this function is used.